libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/notification/RustNotificationService.kt

```diff
@@ -45,7 +45,7 @@ class RustNotificationService(
             }
             val items = notificationClient.getNotifications(requests)
             buildMap {
-                val eventIds = requests.flatMap { it.eventIds }
+                val eventIds = requests.flatMap { it.eventIds }.distinct()
```

Comment:
Oh, very good catch! I tried to fix this before by making sure we didn't have several `.use { ... }` blocks for the same object, but it never occurred to me this could be another source for the issue. In any case, it seems like a sensible check, so LGTM.